src/python/zquantum/qaoa/farhi_ansatz.py

```diff
@@ -22,7 +22,7 @@ def __init__(
         self,
         number_of_layers: int,
         cost_hamiltonian: Union[QubitOperator, IsingOperator],
-        mixer_hamiltonian: Optional[Union[QubitOperator, IsingOperator]] = None,
+        mixer_hamiltonian: Optional[QubitOperator] = None,
```

Comment:
How come the `mixer_hamiltonian` should not be an `IsingOperator`? I assume this is probably trivial, but want to make sure this is the intended behavior.

Reply:
Actually Mixer should never be Ising, because mixer and cost cannot commute.

Reply:
Oh right, and the cost hamiltonian should only have `Z` terms